### DISCUSSION TITLE
Searching in Sumatra

### DISCUSSION THREAD

**Discussion by rogerbratseth** [sumatrapdfreader/sumatrapdf#3217] [Q&A]
I was looking for a way to search for whole words and discovered that I can include space before and after the search-string. But when searching for ' ifs ', as an example, I get a match on 'IFS' immediately followed by an '='-sign, and that is not desirable. Any work-around for this?

When doing the same search in Notepad++ I do not get matches on 'IFS=' when using leading and trailing space in search-string.

Kind regards,
Roger Bratseth

**Comment by GitHubRulesOK** [comment]
pdf is independent glyphs so searching for plain text is problematic the page could be drawn as all letter a then all letter b then all letter c so finding a b c in heuristic terms needs find the first a and if its got a b to the right that's ab then if there is a c its abc but if there is slightly bigger white space between them it's a match for `a b c` the issue here is white space can be anything such as kerning, 3 different types of physical line feed, zero space assignment, nuls or even a simple spacebar character. so varies from source to source, hence many open issues about simple search. A word inside a PDF can be like this `<00480047004C005700480047> Tj`

searching for a simple spacebar character is naturally ignored but `a ` is respected likewise any non a-z termination symbol so `a=` `a/` `a.` `a,` and some others count as the same match

there is no "look ahead" it is just the first match, so for the example ABCD then A is a match AB is a match ab is a match AbC is a match and CD t is a match same as here 
![image](https://user-images.githubusercontent.com/12754913/212049998-1b0f07f4-55de-4677-bd1a-f88da9835952.png)

**Reply by rogerbratseth** [reply]
Thanks for taking your time to answer in such great detail. I had no idea that pdfs were so hard to work with!
In my case it was an epub-document, which might be easier to search? It is html?

**Reply by GitHubRulesOK** [reply]
yes epub is xhtml in zip folder/container (similar to a docx) the letter usage is more akin to web layout even allowing flow between pages but even so that could in rare cases be broken text rather than what we consider as "words"

AFAIK SumatraPDF uses one basic general set of rules for text search (except some specifics for more eastern localities)
